### ISSUE TITLE
onnx generation

### ISSUE THREAD

**Issue by Jillian-Wang** [xjsxujingsong/FairMOT_TensorRT_C#6]
> How to Run
export onnx model file in FairMOT by adding the following code in line 479 in "/src/lib/models/networks/pose_dla_conv.py", and put it to the folder "models"

    z = {}
    for head in self.heads:
        z[head] = self.__getattr__(head)(y[-1])

    hm = z["hm"]
    wh = z["wh"]
    reg = z["reg"]
    hm = F.sigmoid(hm)
    hm_pool = F.max_pool2d(hm, kernel_size=3, stride=1, padding=1)

    id_feature = z['id']
    id_feature = F.normalize(id_feature, dim=1)
    id_feature = id_feature.permute(0, 2, 3, 1).contiguous() #switch id dim
    return [hm, wh, reg, hm_pool, id_feature]

Hi, thanks for your outstanding work, could you please tell me the reason why you modified the return from original [z] ( [hm, wh, reg, id_feature]) to current return [hm, wh, reg, hm_pool, id_feature]
thank your very much.

**Comment by xjsxujingsong** [comment]
You need to find the location of max value of hm.

**Comment by Jillian-Wang** [comment]
> You need to find the location of max value of hm.

May I take it that you don't want to process the "hm_pool = F.max_pool2d(hm, kernel_size=3, stride=1, padding=1)" in tensorrt, instead you get the information directly from python code by model return? because even when we get the  ( [hm, wh, reg, id_feature])  we still need to to "_nms(heat)", you just split the operations apart to several sub operations? Thank you very much

**Comment by Jillian-Wang** [comment]
> export onnx model file in FairMOT by adding the following code in line 479 in "/src/lib/models/networks/pose_dla_conv.py", and put it to the folder "models"

Sorry to bother you again, I have another question if we modified the return from ( [hm, wh, reg, id_feature])  to [hm, wh, reg, hm_pool, id_feature], we need to retrain the model right? we can't use the already trained model([hm, wh, reg, id_feature]) to do the inference to return [hm, wh, reg, hm_pool, id_feature] right?

**Comment by xjsxujingsong** [comment]
The first question: yes.
The second question. You dont have to retrain the model. Just load the pre-trainied model and it will be fine since we dont change the network parameters (in init function), we only change the forward function.

**Comment by Jillian-Wang** [comment]
Oh, I see, Thank you very much! :)

**Comment by Jillian-Wang** [comment]
Hi, sorry to bother you again but when I exported the pytorch model of "https://github.com/zengwb-lx/Face-Tracking-usingFairMOT" to onnx with your guidance and used your this project to load the onnx and parse and run, the project can run but the result is a little strange, the bbox is on the left top of actual face, while the pytorch model of FaceFairMOT can run normally, I checked the decode process of pytorch model return of FaceFairMOT which is the same as original FairMOT, the author only modify the backbone. Could you give me some advice on finding the reason why this happened? I really appreciate that.

**Comment by xjsxujingsong** [comment]
I guess it is because the latest FairMOT changed the output format: 
check this line in src/FairMOT.cpp: detconfig.fairmot.ltrb = true;
You may change it to "false" which is used in the original FairMOT.
You can try the model I uploaded in this repo and make sure it is fine.

**Comment by Jillian-Wang** [comment]
I use np.testing.asser_allclose() to check the output of pytorch model and onnx model and it passed. so I think the issue should be generated by the tensorRT part, I read your code and modify the frame convert part just use cv::resize directly from "frame" to "flt_img", and the result showed normally,  but when the video is playing some bboxes start to shift, I am working on this issue now.

**Comment by Jillian-Wang** [comment]
> I guess it is because the latest FairMOT changed the output format:
check this line in src/FairMOT.cpp: detconfig.fairmot.ltrb = true;
You may change it to "false" which is used in the original FairMOT.
You can try the model I uploaded in this repo and make sure it is fine.

Thanks for your reply， your code with your model runs normally, and I checked the FaceFairMOT in which the opt.ltrb is set "True" in default that is same as yours. I will continue try to find out the reason. :)

**Comment by Jillian-Wang** [comment]
Hi, I find the way of transformation of original frame to model input frame is really important, I used the same function as python function letterbox to do the frame resize , and the result showed normally. Thank you very much.
